resteasy-jaxrs/src/main/java/org/jboss/resteasy/core/StringParameterInjector.java

```diff
@@ -80,40 +80,39 @@ protected void initialize(Class type, Type genericType, String paramName, Class
       baseType = type;
       baseGenericType = genericType;
 
-      if (type.isArray()) baseType = type.getComponentType();
-      if (List.class.isAssignableFrom(type))
-      {
-         isCollection = true;
-         collectionType = ArrayList.class;
-      }
-      else if (SortedSet.class.isAssignableFrom(type))
-      {
-         isCollection = true;
-         collectionType = TreeSet.class;
-      }
-      else if (Set.class.isAssignableFrom(type))
+      if (type.isArray())
       {
-         isCollection = true;
-         collectionType = HashSet.class;
+    	  baseType = type.getComponentType();
       }
-      if (isCollection)
+     
+      if (!baseType.isPrimitive())
       {
-         if (genericType != null && genericType instanceof ParameterizedType)
+         paramConverter = factory.getParamConverter(baseType, baseGenericType, annotations);
+         if (paramConverter != null) 
          {
-            ParameterizedType zType = (ParameterizedType) genericType;
-            baseType = Types.getRawType(zType.getActualTypeArguments()[0]);
-            baseGenericType = zType.getActualTypeArguments()[0];

```

Comment:
Hey Nicolas,

I've been digging a little deeper and re-thinking this issue. I think I understand now why StringParameterInjector.initialize() is written as it is. Basically, it expects the underlying URL parsing mechanisms to handle Collections, and then the various string translation mechanisms to handle the element type. For example, given the resource method

>		@GET
>		@Path("dates")
>		@Produces("text/plain")
>		public String dates(@QueryParam("q") List<Date> q)
>		{
>			for (Date d : q)
>			{
>				System.out.println(d);
>			}
>			return "ok";
>		}

running

>		String date1 = new Date(117, 1, 1).toString();
>		String date2 = new Date(117, 2, 2).toString();
>		ResteasyClient client = new ResteasyClientBuilder().build();
>		Invocation.Builder request = client.target("http://localhost:8081/dates?q=" + date1 + "&q=" + date2).request();
>		Response response = request.get();

will produce

Wed Feb 01 00:00:00 EST 2017
Thu Mar 02 00:00:00 EST 2017

That is, "...?q=a&q=b" can be turned into a list or a set, so maybe we don't really need another syntax.

-Ron